### PR TITLE
Pass minify config to jsminify task.

### PR DESCRIPTION
**Pull request by ghost**
I would like to pass "mangle" and "squeeze" option to jsminify task. The change here allowing mojito app to have such config in application.json:

[
&nbsp;&nbsp;"settings": ["master"],
&nbsp;&nbsp;"shaker": {
&nbsp;&nbsp;&nbsp;&nbsp;"minify": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"config": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"mangle": true,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"squeeze": true
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;}
]

As per my test on our app, this help reduce script size from ~310K to ~220K in our case.

### CODE REVIEW

File: lib/shaker.js [hunk 1]
```diff
@@ -673,7 +673,7 @@ Shaker.prototype = {
             js: {task:['readResources', jsList]},
             compileJS: {requires: 'js', task: ['preprocess']},
             jsLint: {requires: 'compileJS', task: [shakerOptions.jslint ? 'jslint' : 'noop', {callback: jsLintCheck }]},
-            jsMinify: {requires: 'jsLint', task: [shakerOptions.minify? 'jsminify': 'noop']},
+            jsMinify: {requires: 'jsLint', task: (shakerOptions.minify? ['jsminify', shakerOptions.minify]: 'noop'},
```
Comment (caridy) [inline lib/shaker.js]: 1. Did you tried this code? It looks like there is a missing ) in this line. This you run this thru JSLint?
2. when noop should be used, does it need to be an array? I'm not very familiar with the queue implementation.

**Comment by caridy** [comment]
Can you also:
- add a test if posible
- add to the documentation

**Comment by ghost** [comment]
My bad, I didn't test this line. And queue.tasks also accept non-array parameter. See:
https://github.com/twobit/gear/blob/master/lib/tasks/tasks.js#L38-L47

**Comment by ghost** [comment]
I will improve my code to let it pass config to cssminify() as well. Will update pull request later.

**Comment by ghost** [comment]
Updated docs. And now I change config format like this to leave room of css minify options:

[
&nbsp;&nbsp;"settings": ["master"],
&nbsp;&nbsp;"shaker": {
&nbsp;&nbsp;&nbsp;&nbsp;"minify": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"css": null,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"js": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"mangle": true,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"squeeze": true
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;}
]

@caridy Can you show me how to run Shaker tests? So that I can add tests for this change.

**Comment by diervo** [comment]
You can create a folder under test called tasks. Test are standard YUITest files, copy one a follow the convention.
For running a test is just yuitest <filename>.

Shaker does not estrictly run all tests before bumping to a new version, so your pull request should be good.

+1

**Pull request merged**